corehq/apps/accounting/tasks.py

```diff
@@ -810,6 +812,44 @@ def calculate_users_in_all_domains(today=None):
     auto_deactivate_mobile_workers.delay()
 
 
+def count_form_submitting_mobile_workers(domain, start, end):
+    """
+    Returns the count of mobile workers who have submitted a form in the time specified.
+    """
+    form_counts_by_worker = (
+        FormES(for_export=True)
+        .domain(domain)
+        .submitted(gte=start, lt=end)
+        .user_type('mobile')
+        .user_aggregation()
+        .size(0)
+        .run()
+        .aggregations.user.normalized_buckets
+    )
+    return len(form_counts_by_worker)
+
+
+@periodic_task(run_every=crontab(hour=1, minute=0, day_of_month='1'), acks_late=True)
+def calculate_form_submitting_mobile_workers_in_all_domains(today=None):
+    today = today or datetime.date.today()
+    date_start = today - relativedelta(months=1)
+    for domain in Domain.get_all_names():
+        num_workers = count_form_submitting_mobile_workers(domain, date_start, today)
+        record_date = today - relativedelta(days=1)
+        try:
+            FormSubmittingMobileWorkerHistory.objects.create(
+                domain=domain,
+                num_users=num_workers,
+                record_date=record_date
+            )
+        except Exception as e:
```

Comment:
I don't think we're expecting anything to go wrong, but because the object it creates is used directly for invoicing we want to log if anything ever does go wrong. Or at least, that's my understanding of it, since catching the generic `Exception` like this is all over accounting tasks.

In practice, I think the only thing that ever could go wrong is if there is already a record matching that date and domain, which raises `IntegrityError` for violating the uniqueness constraint. I can catch that here instead.

Reply:
I hear you on a catch-all statement for top-level code that is the last point of interaction. But here, this doesn't appear like that, because it's nested inside the domain loop. Given the unreliability of network connections, we could easily see errors occurring during `Domain.get_all_names()` or `count_form_submitting_mobile_workers`, but those errors don't have any error handling. Because of that, it looks to me like we're trying to catch a specific situation with user creation, and in that event, I'd want to be explicit about what errors we're trying to catch

Reply:
I opted to catch the specific exception that would get called if the user history already exists based on the `unique_together` constraint -- which is what would happen if the task got interrupted and restarted partway through, the most plausible situation for this to handle. https://github.com/dimagi/commcare-hq/pull/34868/commits/d7120031b0ec9f029495f322e7977bf660f216f9